lib/linux.js

```diff
@@ -159,5 +173,13 @@ exports.set = function set(imagePath) {
 
 	params[params.length - 1] = params[params.length - 1].replace('%s', path.resolve(imagePath));
 
+	if (app.cmd === 'gsettings') {
+		isCinnamon().then(cinnamon => {
```

Comment:
It's just promises. You can do anything you would do synchronously. You just need to chain the `isCinnamon()` before `cp.execFile` and return the whole chain.